idb/dummy.go

```diff
@@ -217,29 +235,34 @@ type IndexerDb interface {
 
 	LoadGenesis(genesis types.Genesis) (err error)
 	SetProto(version string, proto types.ConsensusParams) (err error)
-	GetProto(version string) (proto types.ConsensusParams, err error)
+	GetProtoTx(tx *sql.Tx, version string) (proto types.ConsensusParams, err error)
 
 	GetImportState() (is *ImportState, err error)
 	SetImportState(ImportState) (err error)
-	GetMaxRoundAccounted() (round uint64, err error)
+	GetMaxRoundAccountedTx(tx *sql.Tx) (round uint64, err error)
 	GetMaxRoundLoaded() (round uint64, err error)
-	GetSpecialAccounts() (SpecialAccounts, error)
+	GetSpecialAccountsTx(ctx context.Context, tx *sql.Tx) (SpecialAccounts, error)
 	GetDefaultFrozen() (defaultFrozen map[uint64]bool, err error)
 
 	// YieldTxns returns a channel that produces the whole transaction stream after some round forward
 	YieldTxns(ctx context.Context, prevRound int64) <-chan TxnRow
 
 	CommitRoundAccounting(updates RoundUpdates, round uint64, blockPtr *types.Block) (err error)
 
-	GetBlock(ctx context.Context, round uint64, options GetBlockOptions) (block types.Block, transactions []TxnRow, err error)
+	GetBlock(ctx context.Context, round uint64,
+		options GetBlockOptions) (block types.Block, transactions []TxnRow, err error)
+	GetBlockTx(ctx context.Context, tx *sql.Tx, round uint64,
+		options GetBlockOptions) (block types.Block, transactions []TxnRow, err error)
```

Comment:
Is there a way to avoid needing 2 versions of each function? Maybe we can check for `tx==nil` and automatically initialize it when needed?